game.go

```diff
@@ -1338,12 +1351,16 @@ func (p *Game) ShowVar(name string) {
 // Widget
 
 // GetWidget returns the widget instance with given name. It panics if not found.
-func (p *Game) GetWidget(name string) Widget {
-	items := p.items
+func Gopt_Game_Gopx_GetWidget[T any](game interface{ Parent() *Game }, name string) *T {
```

Comment:
`interface{ Parent() *Game }` is implemented by `*Sprite`, rather than `*Game`. As a result the method `getWidget` will only be callable in sprite code.

I think we should recieve game instance instead of sprite instance here?

Reply:
![image](https://github.com/user-attachments/assets/8b2640ab-5faa-488b-a8f0-4038ed2b4354)
The above is code generated by Spx, it passes a Sprite pointer instead of a Game pointer.

Reply:
But if we write `getWidget(...)` in the stage code (`main.spx`), the value of `this` will be of type `*Game`.

We expect `getWidget` available both in stage code and sprite code.